core/src/cluster_info.rs

```diff
@@ -94,13 +100,20 @@ pub enum ClusterInfoError {
     BadGossipAddress,
 }
 #[derive(Clone)]
+pub struct DataBudget {
+    bytes: usize,
+    last_timestamp: u64,
```

Comment:
more comments here too please, it's still not obvious to me if more `bytes` is good or bad or what `last_timestamp` is used for